src/librustc/traits/project.rs

```diff
@@ -354,7 +354,8 @@ pub fn normalize_projection_type<'a, 'b, 'gcx, 'tcx>(
             // information is available.
 
             let tcx = selcx.infcx().tcx;
-            let def_id = tcx.associated_items(projection_ty.trait_ref.def_id).find(|i|
+            // FIXME(tschottdorf): simplify?
+            let def_id = tcx.associated_items(projection_ty.trait_ref(selcx.tcx()).def_id).find(|i|
                 i.name == projection_ty.item_name(tcx) && i.kind == ty::AssociatedKind::Type
             ).map(|i| i.def_id).unwrap();
             let ty_var = selcx.infcx().next_ty_var(
```

Comment:
Indeed this is `projection_ty.item_def_id`. In general avoid using the new `trait_ref` method where possible.